src/nal/sps.rs

```diff
@@ -815,25 +834,133 @@ pub struct BitstreamRestrictions {
+            if log2_max_mv_length_horizontal > 16 {
+                return Err(SpsError::FieldValueTooLarge {
+                    name: "log2_max_mv_length_horizontal",
+                    value: log2_max_mv_length_horizontal,
+                });
+            }
+            let log2_max_mv_length_vertical = r.read_ue("log2_max_mv_length_vertical")?;
+            if log2_max_mv_length_vertical > 16 {
+                return Err(SpsError::FieldValueTooLarge {
+                    name: "log2_max_mv_length_vertical",
+                    value: log2_max_mv_length_vertical,
+                });
+            }
+            let max_num_reorder_frames = r.read_ue("max_num_reorder_frames")?;
+            let max_dec_frame_buffering = r.read_ue("max_dec_frame_buffering")?;
+            if max_num_reorder_frames > max_dec_frame_buffering {
+                return Err(SpsError::FieldValueTooLarge {
+                    name: "max_num_reorder_frames",
+                    value: max_num_reorder_frames,
+                });
+            }
+            // "The value of max_dec_frame_buffering shall be greater than or equal to
+            // max_num_ref_frames."
+            if max_dec_frame_buffering < sps.max_num_ref_frames {
+                return Err(SpsError::FieldValueTooLarge {
+                    name: "max_dec_frame_buffering",
+                    value: max_dec_frame_buffering,
+                });
+            }
+            let max = max_val_for_max_dec_frame_buffering(sps);
+            if max_dec_frame_buffering > max {
+                return Err(SpsError::FieldValueTooLarge {
+                    name: "max_dec_frame_buffering",
+                    value: max_dec_frame_buffering,
+                });
+            }
             Some(BitstreamRestrictions {
-                motion_vectors_over_pic_boundaries_flag: r
-                    .read_bool("motion_vectors_over_pic_boundaries_flag")?,
-                max_bytes_per_pic_denom: r.read_ue("max_bytes_per_pic_denom")?,
-                max_bits_per_mb_denom: r.read_ue("max_bits_per_mb_denom")?,
-                log2_max_mv_length_horizontal: r.read_ue("log2_max_mv_length_horizontal")?,
-                log2_max_mv_length_vertical: r.read_ue("log2_max_mv_length_vertical")?,
-                max_num_reorder_frames: r.read_ue("max_num_reorder_frames")?,
-                max_dec_frame_buffering: r.read_ue("max_dec_frame_buffering")?,
+                motion_vectors_over_pic_boundaries_flag,
+                max_bytes_per_pic_denom,
+                max_bits_per_mb_denom,
+                log2_max_mv_length_horizontal,
+                log2_max_mv_length_vertical,
+                max_num_reorder_frames,
+                max_dec_frame_buffering,
             })
         } else {
             None
         })
     }
 }
 
+// calculates the maximum allowed value for the max_dec_frame_buffering field
+fn max_val_for_max_dec_frame_buffering(sps: &SeqParameterSet) -> u32 {
```

Comment:
This introduces panics on unknown level/profile. At least I'd want it to return `Err` instead. What are your thoughts on how important `max_dec_frame_buffering` bounds enforcement is at all? I'm not sure it's necessary to avoid overflows as say the `pic_init_qs_minus26` check is (#58).

Reply:
I left this in a half finished state.

Currently the value would only be in avoiding possible bounds check issues in calling code, since there is no local use of this field.

I'll back this code out in the interests of getting this merged sooner, and it can return at a later date.